packages/vats/src/core/types-ambient.d.ts

```diff
@@ -103,13 +103,16 @@ type ClientProvider = {
 };
 
 type Producer<T> = {
-  resolve: (v: ERef<T>) => void;
+  resolve: (v: ERef<T | import('@endo/exo').Guarded<T>>) => void;
```

Comment:
shouldn't `T` itself by the Guarded type?

Reply:
Good catch!  This was a leftover attempt before I started typing `KindFacet` correctly.  Reverted.